modules/programs/fish.nix

```diff
@@ -343,7 +343,7 @@ in {
         # if we haven't sourced the general config, do it
         if not set -q __fish_general_config_sourced
 
-          set -p fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions
+          set --prepend fish_function_path ${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d
```

Comment:
Then it would need to be `if pkgs ? fish-foreign-env then "${pkgs.fish-foreign-env}/share/fish-foreign-env/functions" else "${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d"`, since the directory providing the function changed as well.

Reply:
`pkgs.fish-foreign-env` still exists (https://github.com/NixOS/nixpkgs/commit/d94921db12d9c9e2d3e9c2e6e25e50f03f789ecb#diff-29ed0f5fda84c91253503c1664d60a194a324e29518d472adf846f30b8db52e3R161), so branching by `pkgs ? fish-foreign-env` seems not to work.

Reply:
Then:

```suggestion
          set --prepend fish_function_path ${if pkgs ? fishPlugins && pkgs.fishPlugins ? foreign-env then "${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d" else "${pkgs.fish-foreign-env}/share/fish-foreign-env/functions"}
```

Reply:
Did anybody try the diff proposed by @cole-h. If it works then we can include it but otherwise I think it should be OK to merge as-is since I think it is reasonable to assume that HM follows Nixpkgs master as closely as possible.

Reply:
I'll report on it later today. I agree simply tracking master is ok but if it's this simple to keep backwards compatibility, maybe it's fine.

Reply:
Wanted to try this out but I realized there's no `home.imports` or similar which is how I replace nixos modules. What's the most straightforward way for me to test this patch?

Reply:
You can set `home.path = ~/path/to/cloned/home-manager` and apply the modification locally.
I don't know of a better way, sorry.

Reply:
Thanks to rycee's help I can now confirm that cole-h's suggestion works. The h-m manual makes a point of mentioning backward-compatibility for modules in this case the price for that is a branch : )